sdk/ml/azure-ai-ml/azure/ai/ml/entities/_builders/data_transfer.py

```diff
@@ -346,7 +355,7 @@ def _picked_fields_from_dict_to_rest_object(cls) -> List[str]:
 
     def _customized_validate(self):
         result = super()._customized_validate()
-        if self.source is None:
+        if not self.source:
```

Comment:
what do you want to handle here? usually `self.source is None` is safer

Reply:
Actually, worry about source here maybe an empty dict. But I will deal with this case in _build_source_sink. 

Update to self.source is None